### PR TITLE
KOGITO-7171: [SWF Editor] GWT-REACT bridge for TextEditorAPI

### PR DESCRIPTION
**Pull request by fantonangeli**
Jira issue: https://issues.redhat.com/browse/KOGITO-7171
@caponetto @tiagobento @yesamer

WIP. I open this PR for review. 

To Be Done:
-implement the communication in vscode

Please let me know if I'm in the right direction.

### CODE REVIEW

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.kie.workbench.common.stunner.sw.client.editor;
+
+import jsinterop.annotations.JsOverlay;
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsType;
+
+@JsType(isNative = true, namespace = JsPackage.GLOBAL, name = "window")
+public class DiagramWindow {
```
[the remark this answers is not in the document]
Comment (yesamer) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java:8]: @fantonangeli I can give you some advice to improve the Bridge, I'll contact you offline

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/pom.xml [hunk 2]
```diff
@@ -528,4 +528,4 @@
     </profile>
   </profiles>
 
-</project>
\ No newline at end of file
+</project>
```
[the remark this answers is not in the document]
Comment (fantonangeli) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/pom.xml:531]: I had a lot of conflicts in the merge. I guess is the resolving of a conflict, when I saved the file and marked as resolved.

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.kie.workbench.common.stunner.sw.client.editor;
```
[the remark this answers is not in the document]
Comment (yesamer) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java:1]: Mmh, Probably all the bridges defined in `appformer-kogito-bridge` are not used by SWF editor. so I suppose you agreed with Tiago or Guilerme to put the bridge there. is that correct? 
In that case, I would argue we may clean up the bridges defined in that module (btw not in this PR of course)

File: packages/serverless-workflow-diagram-editor-envelope/src/envelope/vscode/VsCodeServerlessWorkflowDiagramEditorFactory.ts [hunk 4]
```diff
@@ -17,13 +17,14 @@
...
 import { ServerlessWorkflowDiagramEditor } from "../ServerlessWorkflowDiagramEditor";
-import { ServerlessWorkflowDiagramEditorFactory } from "../ServerlessWorkflowDiagramEditorFactory";
+import { ServerlessWorkflowDiagramEditorFactory, onNodeClicked } from "../ServerlessWorkflowDiagramEditorFactory";
 import { VsCodeServerlessWorkflowDiagramEditorChannelApi } from "./VsCodeServerlessWorkflowDiagramEditorChannelApi";
 
 export interface CustomWindow extends Window {
   envelope: {
     javaCodeCompletionService: JavaCodeCompletionApi;
   };
+  onNodeClicked: (nodeName: string) => void;
```
Comment (yesamer) [inline packages/serverless-workflow-diagram-editor-envelope/src/envelope/vscode/VsCodeServerlessWorkflowDiagramEditorFactory.ts:27]: @tiagobento @caponetto @fantonangeli 
1. Could we introduce an interface that represents a `TextEditorAPI` here? (similarly to the `JavaCodeCompletionAPI`). In this case, where we can define this interface?
2. Should we define the service inside the envelope, I guess?
3. (Not related, my curiosity) Do you plan to use `JavaCodeCompletionAPI` in SWF?

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yesamer) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/editor/DiagramWindow.java:1]: @tiagobento @caponetto @fantonangeli We should understand the correct place for this JSInterop class. I think this bridge will be consumed by SWF only, so putting it here and not in `appformer-kogito-bridge` can be a valid choice. Maybe I would add a `bridge` package at the end.

**Comment by caponetto** [comment]
Closing because the branch is out of sync and @fantonangeli moved changes to a new branch.
A new PR will be opened.

**Pull request closed without merging**